### PR TITLE
Retail utility - added new checks, updated schema

### PR DESCRIPTION
**Pull request by Jagannath-Padhy**
- added missing enums for size 
- removed colour name from enums, added hex codes
- order_id now accepts both uuid and 32 characters alphanumeric id with '-'
- updated on_cancel schema
- removed available.count enums from on_search
- added bpp_terms in schema of on_search
- added multiple checks to handle code break and descriptive logs
- added checks for long and short description
- added error for accept_bap_terms and collect_payment as not required for now
- refactored for better readability
- added schema checks for tags in on_search
- added check for provider_tax_number
- changed available and maximum count from enum to numbers in schema

### CODE REVIEW

File: schema/Retail/RET/on_search.ts [hunk 1]
```diff
@@ -507,9 +507,6 @@ export const onSearchSchema = {
                             },
                             code: {
                               type: 'string',
-                              pattern: '^(1|2|3|4|5):[a-zA-Z0-9]+$',
-                              errorMessage:
```
Comment (sandeepshahi) [inline schema/Retail/RET/on_search.ts:511]: why is this validation removed?

File: schema/Retail/Cancel/on_cancel_rto.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ export const cancel_RTO_Schema = {
             id: {
               type: 'string',
               minLength: 1,
-              pattern: '^[a-zA-Z0-9]{1,32}$',
+              pattern: '^[a-zA-Z0-9-]{1,32}$|^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$',
               errorMessage: 'Should be alphanumeric upto 32 letters max',
```
Comment (sandeepshahi) [inline schema/Retail/Cancel/on_cancel_rto.ts:86]: Can we specify that the error is related to order ID

File: schema/Retail/Update/on_update.ts [hunk 3]
```diff
@@ -75,7 +75,7 @@ export const onUpdateSchema = {
           type: 'object',
           properties: {
             id: { type: 'string',
-            pattern: '^[a-zA-Z0-9]{1,32}$',
+            pattern: '^[a-zA-Z0-9-]{1,32}$|^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$',
             errorMessage: 'Should be alphanumeric upto 32 letters max',
            },
             state: { type: 'string', enum: ['Created', 'Accepted', 'Cancelled', 'Completed', 'Delivered'] },
```
Comment (sandeepshahi) [inline schema/Retail/Update/on_update.ts:81]: In the case of return flow (fulfillment type='Return'), the start location ID will not be present. Instead, the start and end locations are swapped, and the location ID of the buyer does not hold any significance

File: schema/Retail/RET/on_init.ts [hunk 4]
```diff
@@ -528,7 +528,6 @@ export const onInitSchema = {
                 '@ondc/org/settlement_details',
               ],
             },
-
             tags: {
               type: 'array',
               items: {
```
Comment (sandeepshahi) [inline schema/Retail/RET/on_init.ts:533]: provider_tax_number and tax_number must be made mandatory in /tags/bpp_terms

File: schema/Retail/Cancel/on_cancel_rto.ts (hunk 2 above)
Comment (Jagannath-Padhy) [reply schema/Retail/Cancel/on_cancel_rto.ts]: done

File: schema/Retail/RET/on_init.ts (hunk 4 above)
Comment (Jagannath-Padhy) [reply schema/Retail/RET/on_init.ts:533]: done

File: schema/Retail/Update/on_update.ts (hunk 3 above)
Comment (Jagannath-Padhy) [reply schema/Retail/Update/on_update.ts:81]: will address this in next MR ASAP,
can we merge this for now so other merge conflicts dont arise

File: schema/Retail/Update/on_update.ts [hunk 5]
```diff
@@ -75,8 +75,8 @@ export const onUpdateSchema = {
           type: 'object',
           properties: {
             id: { type: 'string',
-            pattern: '^[a-zA-Z0-9]{1,32}$',
-            errorMessage: 'Should be alphanumeric upto 32 letters max',
+            pattern: '^[a-zA-Z0-9-]{1,32}$|^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$',
```
[the remark this answers is not in the document]
Comment (sandeepshahi) [reply schema/Retail/Update/on_update.ts:78]: same is applicable for all the APIs post /confirm

File: schema/Retail/RET11/on_init.ts [hunk 6]
```diff
@@ -538,24 +538,146 @@ export const FnBonInitSchema = {
...
+                        },
+                        {
+                          if: {
+                            properties: {
+                              code: {
+                                const: 'tax_number',
+                              },
+                            },
+                          },
+                          then: {
```
Comment (sandeepshahi) [inline schema/Retail/RET11/on_init.ts:646]: only tax_number and provider_tax_number are required right now in bpp_terms excluding rest of the tnCs

**Review by sandeepshahi — APPROVED** [review]

**Pull request merged**